pkg/frontend/installversions_list.go

```diff
@@ -0,0 +1,37 @@
+package frontend
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/sirupsen/logrus"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/frontend/middleware"
+	"github.com/Azure/ARO-RP/pkg/util/version"
+)
+
+func (f *frontend) listInstallOpenShiftVersions(w http.ResponseWriter, r *http.Request) {
+	vars := mux.Vars(r)
+	log := r.Context().Value(middleware.ContextKeyLog).(*logrus.Entry)
+	versions := f.getInstallOpenShiftVersions()
+
+	converter := f.apis[vars["api-version"]].InstallOpenShiftVersionsConverter()
+
+	b, err := json.Marshal(converter.ToExternal((*api.InstallOpenShiftVersions)(&versions)))
```

Comment:
There is no check for the validity of the returned `converter` and when you call this endpoint with an older unsupported API it results in a panic. This function should look a little more like this:
```go
func (f *frontend) listInstallOpenShiftVersions(w http.ResponseWriter, r *http.Request) {
	ctx := r.Context()
	log := ctx.Value(middleware.ContextKeyLog).(*logrus.Entry)
	vars := mux.Vars(r)

	if f.apis[vars["api-version"]].InstallOpenShiftVersionsConverter == nil {
		api.WriteError(w, http.StatusBadRequest, api.CloudErrorCodeInvalidResourceType, "", "The endpoint could not be found in the namespace '%s' for api version '%s'.", vars["resourceProviderNamespace"], vars["api-version"])
		return
	}

	converter := f.apis[vars["api-version"]].InstallOpenShiftVersionsConverter()
	versions := f.getInstallOpenShiftVersions()

	b, err := json.Marshal(converter.ToExternal((*api.InstallOpenShiftVersions)(&versions)))
	reply(log, w, nil, b, err)
}
```